tests/gordo_components/model/test_model_factories_utils.py

```diff
@@ -1,25 +1,55 @@
 # -*- coding: utf-8 -*-
 
-import unittest
-
-from gordo_components.model.factories.model_factories_utils import hourglass_calc_dims
-
-
-class ModelFactoriesUtilsTestCase(unittest.TestCase):
-    def test_hourglass_calc_dims_check_dims(self):
-        """
-        Test that hourglass_calc_dims implements the correct dimensions
-        """
-
-        dims = hourglass_calc_dims(0.2, 4, 5)
-        self.assertEqual(dims, [4, 3, 2, 1])
-        dims = hourglass_calc_dims(0.5, 3, 10)
-        self.assertEqual(dims, [8, 7, 5])
-        dims = hourglass_calc_dims(0.5, 3, 3)
-        self.assertEqual(dims, [3, 2, 2])
-        dims = hourglass_calc_dims(0.3, 3, 10)
-        self.assertEqual(dims, [8, 5, 3])
-        dims = hourglass_calc_dims(1, 3, 10)
-        self.assertEqual(dims, [10, 10, 10])
-        dims = hourglass_calc_dims(0, 3, 100000)
-        self.assertEqual(dims, [66667, 33334, 1])
+import pytest
+
+from gordo_components.model.factories.model_factories_utils import (
+        hourglass_calc_dims,
+        _check_if_only_dim_or_func,
+        _check_dim_func_len,
+        set_dim_func
+        )
+
+
+def test_hourglass_calc_dims_check_dims():
+    """
+    Test that hourglass_calc_dims implements the correct dimensions
+    """
+
+    dims = hourglass_calc_dims(0.2, 4, 5)
+    assert dims == [4, 3, 2, 1]
+    dims = hourglass_calc_dims(0.5, 3, 10)
+    assert dims == [8, 7, 5]
+    dims = hourglass_calc_dims(0.5, 3, 3)
+    assert dims == [3, 2, 2]
+    dims = hourglass_calc_dims(0.3, 3, 10)
+    assert dims == [8, 5, 3]
+    dims = hourglass_calc_dims(1, 3, 10)
+    assert dims == [10, 10, 10]
+    dims = hourglass_calc_dims(0, 3, 100000)
+    assert dims == [66667, 33334, 1]
```

Comment:
Nice, but I think this is a prime candidate for `input, expected` parameterization.

Reply:
This was the one test method for the pre-existing utils module that I put my routines in. Should I rewrite things like this as I come across them and include them in my PR, or maybe create a new issue/PR with things I uncover as I work on something else?

Reply:
I've included this in this PR, but just let me know if I should pull it out to a different PR.

Reply:
ya, just because it's a pre-existing condition doesn't mean we should repeat it. ;-)